src/test/java/org/springframework/integration/kafka/outbound/OutboundTests.java

```diff
@@ -76,9 +82,42 @@ public static void tearDown() {
 
 	@Test
 	public void testAsyncProducerFlushed() throws Exception {
-		KafkaConsumerContext<String, String> consumerContext = createConsumer();
-		// pre-consume to start the receiver because the high-level API doesn't support --from-beginning
-		consumerContext.receive();
+
+		// create the topic
+
+		try {
+			AdminUtils.createTopic(kafkaRunning.getZkClient(), TOPIC, 1, 1, new Properties());
+		}
+		catch (TopicExistsException e) {
+			// do nothing
+		}
+
+		final String suffix = UUID.randomUUID().toString();
+
+		ZookeeperConfiguration configuration = new ZookeeperConfiguration(new ZookeeperConnect());
+		DefaultConnectionFactory connectionFactory = new DefaultConnectionFactory(configuration);
+		connectionFactory.afterPropertiesSet();
+		final KafkaMessageListenerContainer kafkaMessageListenerContainer = new KafkaMessageListenerContainer(connectionFactory, TOPIC);
```

Comment:
This test is failed for me here, because there is no partitions for the provided topic: the `DefaultConnectionFactory.partitionBrokerMapReference.get()` doesn't contains any `partitionsByTopic`, becuase `DefaultConnectionFactory#refreshLeaders` tries to fetch `topicsMetadata` jut only for the `defaultTopic`, which is `null`.

Have I missed anything there, or is it just a bug in the robustness between components?